Rules/UseProcessBlockForPipelineCommand.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    public class UseProcessBlockForPipelineCommand : IScriptRule
+    {
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            IEnumerable<Ast> functionAsts = ast.FindAll(testAst => testAst is FunctionDefinitionAst, true);

```

Comment:
```suggestion
            IEnumerable<Ast> functionAsts = ast.FindAll(testAst => testAst is FunctionDefinitionAst, searchNestedScriptBlocks: true);
```

Reply:
Currently this applies only the function definitions, but the same should hold for any scriptblock (defined by a `ScriptBlockAst`, which is the type of `FunctionDefinitionAst.Body`). Maybe this rule could be generalised?

Reply:
@rjmholt I think you're right! I'll take a look at that soon to see how easy it would be to implement.

Reply:
Implemented in #707f02d96ac5763afaf0b69649d787f8f25b2492.

I think the only possible change that might need to be made is to update the strings file to reflect "script" or "scriptblock" instead of "function." Thoughts?

Reply:
I think a simple change like that makes sense. It would also be possible to parameterise the resource string to use a given syntax name and then decide that based on the parent AST type, but there could be various edge cases there and PowerShell doesn't guarantee that the parent AST's type will remain the same for the same syntax across PowerShell versions.

Reply:
Well, i can't seem to push changes to the strings.resx to save my life 🤣 

Even though on commit the diff is just the lines I changes, it ends up diffing the whole file. I've checked my git config and don't see anything that I believe would cause this. I'll keep digging on that.

Reply:
Are you using Visual Studio? It usually rewrites the whole file. Best to edit the resx in VSCode or another dumber editor

Reply:
Otherwise it might be encoding or line endings. Not sure what this repo's standard is for either, but should be simple enough to work out

Reply:
Yeah I’m using vs 2019, but when I commit it just appears to show the changes, then when I push the whole file is diffed.

Reply:
I'll try changing again in VS Code. Really the only thing I think I would change replacing the one instance of the work function with "command."

Reply:
So looking at the `New-StronglyTypedCsFileForResx.ps1` script and the comment at the top says:

```powershell
<#
# This script can be used to update the *.Designer file if a *.resx ressource file has been updated.
# However, it is recommended to use Visual Studio instead for editing ressources instead since it takes of that automatically and prodcues cleaner diffs.
#>
```

Weird.

Reply:
Yeah, so you want to edit the resx in a simple editor, but let Visual Studio do the C# generation from the resx.

VSCode has better encoding/git support, but VS has the tool for resx -> C# conversion.

Reply:
From the diff this looks like what you've done already. You can't help the massive diff with the Designer.cs file. We basically treat it as a binary since it's autogenerated. You'll see similar diffs in any PRs that touch the resx file in this repo.

Reply:
Ok, so that worked. a7ed1228 has the updated consistent language and didn't diff the whole file.